receiver/sqlqueryreceiver/scraper.go

```diff
@@ -67,7 +70,7 @@ func (s scraper) Scrape(ctx context.Context) (pmetric.Metrics, error) {
 	var errs error
 	for _, metricCfg := range s.query.Metrics {
 		for i, row := range rows {
-			if err = rowToMetric(row, metricCfg, ms.AppendEmpty()); err != nil {
+			if err = rowToMetric(row, metricCfg, ms.AppendEmpty(), ts); err != nil {
```

Comment:
One `ts` value should be used for all datapoint within Scrape call. You pass the correct value for timestamp, but `rowToMetric` uses it as start timestamp

Reply:
You can just fix timestamp only in this PR, I think this should be enough to resolve the immediate need. Start timestamp can be handled separately after that